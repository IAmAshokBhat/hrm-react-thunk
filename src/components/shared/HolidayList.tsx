import Paper from '@mui/material/Paper';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableCell';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';

export const HolidayList = () => {
  const data = [
    {
      date: '01/01/2022',
      festivalName: 'New Years day'
    },
    {
      date: '14/01/2022',
      festivalName: 'Sankranthi'
    },
    {
      date: '26/01/2022',
      festivalName: 'Republic Day'
    },
    {
      date: '17/03/2022',
      festivalName: 'Holi'
    },
    {
      date: '15/04/2022',
      festivalName: 'Good Friday'
    },
    {
      date: '03/05/2022',
      festivalName: 'Khutba - E - Ramzan'
    },
    {
      date: '15/08/2022',
      festivalName: 'Independence Day'
    },
    {
      date: '31/08/2022',
      festivalName: 'Ganesha Festival'
    },
    {
      date: '05/10/2022',
      festivalName: 'Dasara / Vijayadashmi'
    }
  ];
  return (
    <TableContainer component={Paper} width="100%">
      <Table sx={{ minWidth: 340 }} aria-label="holiday table" size="small">
        <TableHead>
          <TableRow>
            <TableCell>Date</TableCell>
            <TableCell>Festival Name</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {data.map(({ festivalName, date }) => (
            <TableRow
              key={festivalName}
              sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
              <TableCell component="th" scope="row">
                {date}
              </TableCell>
              <TableCell component="th" scope="row">
                {festivalName}
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};
