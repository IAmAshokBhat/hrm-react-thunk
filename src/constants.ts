export interface ILeave {
  leaveType: string;
  from: Date;
  to: Date;
  status: string;
  noOfDays: number;
}

export interface ILeaveInfo {
  from: string;
  to: string;
  category: string;
}

export interface ISelectLeaveType {
  value: number;
  label: string;
  balance: number;
}

export enum ACTIONS {
  GET_USERS = 'GET_USERS',
  GET_USERS_LOADING = 'GET_USERS_LOADING',
  GET_USERS_SUCCESS = 'GET_USERS_SUCCESS',
  GET_USERS_FAILURE = 'GET_USERS_FAILURE',

  GET_USER_DETAILS = 'GET_USER_DETAILS',
  GET_USER_DETAILS_LOADING = 'GET_USER_DETAILS_LOADING',
  GET_USER_DETAILS_SUCCESS = 'GET_USER_DETAILS_SUCCESS',
  GET_USER_DETAILS_FAILURE = 'GET_USER_DETAILS_FAILURE',

  GET_ATTENDANCE_DETAILS = 'GET_ATTENDANCE_DETAILS',
  GET_ATTENDANCE_DETAILS_LOADING = 'GET_ATTENDANCE_DETAILS_LOADING',
  GET_ATTENDANCE_DETAILS_SUCCESS = 'GET_ATTENDANCE_DETAILS_SUCCESS',
  GET_ATTENDANCE_DETAILS_FAILURE = 'GET_ATTENDANCE_DETAILS_FAILURE',

  GET_LEAVE_BALANCE = 'GET_LEAVE_BALANCE',
  GET_LEAVE_BALANCE_LOADING = 'GET_LEAVE_BALANCE_LOADING',
  GET_LEAVE_BALANCE_SUCCESS = 'GET_LEAVE_BALANCE_SUCCESS',
  GET_LEAVE_BALANCE_FAILURE = 'GET_LEAVE_BALANCE_FAILURE',

  GET_EVENT_DETAILS = 'GET_EVENT_DETAILS',
  GET_EVENT_DETAILS_LOADING = 'GET_EVENT_DETAILS_LOADING',
  GET_EVENT_DETAILS_SUCCESS = 'GET_EVENT_DETAILS_SUCCESS',
  GET_EVENT_DETAILS_FAILURE = 'GET_EVENT_DETAILS_FAILURE',

  GET_ALL_LEAVES = 'GET_ALL_LEAVES',
  GET_ALL_LEAVES_LOADING = 'GET_ALL_LEAVES_LOADING',
  GET_ALL_LEAVES_SUCCESS = 'GET_ALL_LEAVES_SUCCESS',
  GET_ALL_LEAVES_FAILURE = 'GET_ALL_LEAVES_FAILURE',

  GET_APPLY_LEAVE = 'GET_APPLY_LEAVE',
  GET_APPLY_LEAVE_LOADING = 'GET_APPLY_LEAVE_LOADING',
  GET_APPLY_LEAVE_SUCCESS = 'GET_APPLY_LEAVE_SUCCESS',
  GET_APPLY_LEAVE_FAILURE = 'GET_APPLY_LEAVE_FAILURE',

  GET_ADD_ATTENDANCE = 'GET_ADD_ATTENDANCE',
  GET_ADD_ATTENDANCE_LOADING = 'GET_ADD_ATTENDANCE_LOADING',
  GET_ADD_ATTENDANCE_SUCCESS = 'GET_ADD_ATTENDANCE_SUCCESS',
  GET_ADD_ATTENDANCE_FAILURE = 'GET_ADD_ATTENDANCE_FAILURE',

  GET_LOGIN = 'GET_LOGIN',
  LOGIN_LOADING = 'GET_LOGIN_LOADING',
  LOGIN_SUCCESS = 'GET_LOGIN_SUCCESS',
  LOGIN_FAILURE = 'GET_LOGIN_FAILURE',

  GET_DESIGNATION_LIST = 'GET_DESIGNATION_LIST',
  GET_DESIGNATION_LIST_LOADING = 'GET_DESIGNATION_LIST_LOADING',
  GET_DESIGNATION_LIST_SUCCESS = 'GET_DESIGNATION_LIST_SUCCESS',
  GET_DESIGNATION_LIST_FAILURE = 'GET_DESIGNATION_LIST_FAILURE',

  REGISTER = 'REGISTER',
  REGISTER_LOADING = 'REGISTER_LOADING',
  REGISTER_SUCCESS = 'REGISTER_SUCCESS',
  REGISTER_FAILURE = 'REGISTER_FAILURE'
}

export interface IUserAPI {
  user_id: number;
  name: string;
}
export interface IUserListAPI {
  data: IUserAPI[];
}

export interface IUserInfoDataAPI {
  user_id: number;
  name: string;
  designation: string;
  manager: string;
  email: string;
}

export interface IUserInfoAPI {
  data: IUserInfoDataAPI[];
}

export interface IUserInfo {
  userId: number;
  name: string;
  designation: string;
  manager: string;
  email: string;
}

interface ILeaveBalanceItemAPI {
  leave_type_id: number;
  leave_type: string;
  balance: number;
}

export interface ILeaveBalanceAPI {
  data: ILeaveBalanceItemAPI[];
}

export interface ILeaveBalance {
  leaveTypeId: number;
  leaveType: string;
  balance: number;
}

interface IAttendanceItemAPI {
  name: string;
  attendance_date: string;
  start_time: string;
  end_time: string;
}

export interface IAttendanceDetailsAPI {
  data: IAttendanceItemAPI[];
}

export interface IAttendanceItem {
  name: string;
  attendanceDate: string;
  startTime: string;
  endTime: string;
  noOfHours: number;
}

export interface IError {
  ok: boolean;
  message: string;
  status: number;
}
// eslint-disable-next-line @typescript-eslint/no-explicit-any
interface IAsyncState<T = any> {
  error: IError | null;
  loading: boolean;
  value: T;
}

export interface IDefaultRoot {
  userList: IAsyncState<ILabelValue[]>;
  userInfo: IAsyncState<IUserInfo | null>;
  leaveBalance: IAsyncState<ILeaveBalance[]>;
  attendanceDetails: IAsyncState<IAttendanceItem[]>;
  evetDetails: IAsyncState<IEvent[]>;
  allLeaves: IAsyncState<ILeave[]>;
  leaveCountByMonth: IAsyncState<ILeaveCountByMonth[] | []>;
  applyLeave: IAsyncState<IGenericResponse>;
  addAttendance: IAsyncState<IGenericResponse>;
  loginDetails: IAsyncState<ILoginAPI>;
  designationList: IAsyncState<ILabelValue[]>;
  registeration: IAsyncState<IGenericResponse>;
}

export interface IAction {
  type: string;
}

export interface IUserListAction extends IAction {
  payload: IUserAPI[] | null;
}

export interface IUserInfoAction extends IAction {
  payload: IUserInfoDataAPI | null;
}

export interface ILeaveBalanceAction extends IAction {
  payload: ILeaveBalanceItemAPI[] | null;
}

export interface IAttendanceDetailsAction extends IAction {
  payload: IAttendanceItemAPI[] | null;
}

export interface IEventDetailsAction extends IAction {
  payload: IEventItemAPI[] | null;
}

interface ILeaveItemAPI {
  from_date: string;
  to_date: string;
  leave_type: string;
  status: number;
}

export interface IAllLeavesAction extends IAction {
  payload: ILeaveItemAPI[] | null;
}

export interface IAllLeavesAPI {
  data: ILeaveItemAPI[];
}

//TODO
export type TFAttendanceDetailsAction = (
  id: number,
  from: string,
  to: string
) => Promise<Response>;

interface IEventItemAPI {
  attendance_date?: string;
  start_time?: string;
  end_time?: string;
  from_date?: string;
  to_date?: string;
  leave_type?: string;
}

export interface IEventDetailsAPI {
  data: IEventItemAPI[];
}

export const DATE_FORMAT = 'DD-MM-YYYY';
export const DATE_FORMAT_CALENDAR = 'YYYY-MM-DD';

export enum EventType {
  WORKING = 'WORKING',
  LEAVE = 'LEAVE'
}

export interface IEvent {
  title: string;
  start: string;
  end: string;
  allDay?: boolean;
}

export interface IKeyValue {
  [key: string]: number;
}

export interface ILeaveCountByMonth {
  days: number;
  month: string;
}

export interface INewLeave {
  leaveTypeId: number;
  fromDate: string;
  toDate: string;
  userId: number;
}

export interface IGenericResponse<T = unknown> {
  status: number;
  message: string;
  data: T;
}

export interface IGenericAction extends IAction {
  payload: IGenericResponse | null;
}

export const enum NotificationType {
  SUCCESS = 'success',
  ERROR = 'error',
  WARNING = 'warning',
  INFO = 'info'
}

export interface INewAttendance {
  userId: number;
  date: string;
  inTime: string;
  outTime: string;
}

export interface ILoginData {
  userId: number;
  password: string;
}

export interface ILoginAPI {
  token: string;
  issued: number;
  expires: number;
  loginStatus: number;
  message: string;
  userId: number;
}

export interface ILoginAction extends IAction {
  payload: ILoginAPI | null;
}

export interface IRegistration {
  name: string;
  password: string;
  confirmPassword: string;
  email: string;
}

export interface ILabelValue {
  value: number;
  label: string;
}

export interface IDesignationListAPI {
  data: ILabelValue[];
}

export interface IDesignationListAction extends IAction {
  payload: ILabelValue[] | null;
}

export interface IRegisterAPI {
  name: string;
  designation: number;
  manager: number;
  password: string;
  email: string;
}

export interface IRegisterAction extends IAction {
  payload: IRegisterAPI | null;
}

export type ExpirationStatus = 'expired' | 'active' | 'grace';

export interface Session {
  id: number;
  dateCreated: number;
  username: string;
  issued: number;
  expires: number;
}

export type DecodeResult =
  | {
      type: 'valid';
      session: Session;
    }
  | {
      type: 'integrity-error';
    }
  | {
      type: 'invalid-token';
    };

export type TAnyObject = { [key: string]: any }; // eslint-disable-line @typescript-eslint/no-explicit-any

export type TAny = any; // eslint-disable-line @typescript-eslint/no-explicit-any
