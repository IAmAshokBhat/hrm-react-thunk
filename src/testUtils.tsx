import { render, RenderOptions, waitFor } from '@testing-library/react';
import { SnackbarProvider } from 'notistack';
import { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { applyMiddleware, createStore, Store } from 'redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { IDefaultRoot, TAny, TAnyObject } from './constants';
import { TFAction } from './contracts';
import { reducers } from './redux/reducers';
import {
  defaultArrayState,
  defaultNullState,
  defaultState
} from './redux/reducers/__tests__/fixtures';

export const middlewares = [thunk];

export const createStoreWithMiddlewares = (
  initialState: TAnyObject = {}
): Store => {
  return createStore(reducers, initialState, applyMiddleware(...middlewares));
};
interface IRenderWithProviderOptions {
  initialStoreState?: TAnyObject;
  renderFunction?: TAny;
  options?: RenderOptions;
  route?: string;
  path?: string;
}

export const renderWithProvider = (
  ui: ReactElement,
  renderOptions?: IRenderWithProviderOptions
) => {
  const {
    initialStoreState = {},
    renderFunction = render,
    options = {},
    route = '',
    path = ''
  } = renderOptions || {};

  const appStore = createStoreWithMiddlewares(initialStoreState);

  let wrappedUi = (
    <SnackbarProvider
      anchorOrigin={{
        vertical: 'top',
        horizontal: 'right'
      }}
    >
      <Provider store={appStore}>{ui}</Provider>
    </SnackbarProvider>
  );

  if (route) {
    window.history.pushState({}, '', route);

    wrappedUi = (
      <Provider store={appStore}>
        <BrowserRouter>
          <Routes>
            <Route path={path} element={ui} />
          </Routes>
        </BrowserRouter>
      </Provider>
    );
  }

  return renderFunction(wrappedUi, options);
};

type TFApiCall = (...params: TAny) => Promise<Response>;

export interface IResponseMock {
  headers?: object;
  ok?: boolean;
  status?: number;
  statusText?: string;
  type?: ResponseType;
  url?: string;
  body?: TAny | null;
  bodyUsed?: boolean;
  blob?: () => Promise<TAny>;
}

const dummyPromise = () => new Promise((resolve) => resolve('dummy'));

export class ResponseMock {
  headers: object;

  ok: boolean;

  status: number;

  statusText: string;

  type: ResponseType;

  url: string;

  body: ReadableStreamDefaultReadResult<Uint8Array> | null;

  bodyUsed: boolean;

  blob: () => Promise<TAny>;

  constructor(props: IResponseMock = {}) {
    this.headers = props.headers || {};
    this.status = props.status || 200;
    this.ok = props.ok || this.status === 200;
    this.statusText = props.statusText || 'OK';
    this.type = props.type || 'basic';
    this.url = props.url || 'http://localhost';
    this.body = props.body;
    this.bodyUsed = props.bodyUsed || this.body !== null;
    this.blob = props.blob || dummyPromise;
  }

  json: () => Promise<TAny> = () => {
    const body = this.body || '""';

    try {
      return Promise.resolve(JSON.parse(body as string));
    } catch {
      return Promise.resolve(body);
    }
  };

  text: () => Promise<string> = () => Promise.resolve(this.statusText);
}

export const mockAPICall = (func: TFApiCall, response: IResponseMock) =>
  (func as jest.Mock).mockResolvedValue(new ResponseMock(response));

export const generateActionTests = async (
  actionName: string,
  action: TFAction,
  api: TAny,
  actionPayload: TAny = []
) => {
  const response = new ResponseMock({ ok: true, body: { data: [] } });

  const mockStore = configureStore(middlewares);

  (api as jest.Mock).mockReturnValue(Promise.resolve(response));
  const store = mockStore({});

  //TODO: Fix TS error
  // @ts-ignore
  await store.dispatch(action);

  await waitFor(() => {
    expect(store.getActions()).toEqual([
      {
        type: `${actionName}_LOADING`
      },
      {
        type: `${actionName}_SUCCESS`,
        //TODO: Accept action params from function params
        payload: actionPayload
      }
    ]);
  });
};

export const generateReducerTests = (
  reducerName: string,
  reducer: TAny,
  selector: TAny,
  actionName: string,
  successValue?: TAny,
  defaultValue?: TAny,
  expectedValue?: TAny,
  reducerLabel?: string
) => {
  describe(`for ${reducerName}`, () => {
    const reducerLbl = reducerLabel
      ? reducerLabel
      : reducerName.replace('Reducer', '');
    it(`${reducerName} reducer loading`, () => {
      reducer(undefined, {
        type: `${actionName}_LOADING`,
        payload: null
      });
      // TODO: Check why its failing
      // expect(selector(defaultState).loading).toEqual(true);
      expect(selector(defaultState).value).toEqual(
        defaultValue === null ? defaultValue : []
      );
      expect(selector(defaultState).error).toEqual(null);
    });

    it(`${reducerName} reducer success`, () => {
      const successState: IDefaultRoot = {
        ...defaultState,
        [reducerLbl]: {
          ...defaultNullState,
          value: expectedValue ? expectedValue : successValue
        }
      };
      reducer(defaultArrayState, {
        type: `${actionName}_SUCCESS`,
        payload: successValue
      });

      expect(selector(successState).error).toEqual(null);
      expect(selector(successState).loading).toEqual(false);
      expect(selector(successState).value).toEqual(
        expectedValue ? expectedValue : successValue
      );
    });

    it(`${reducerName} reducer error`, () => {
      const error = { message: '', status: 1, ok: false };
      const failedState: IDefaultRoot = {
        ...defaultState,
        [reducerLbl]: {
          ...defaultNullState,
          value: [],
          error
        }
      };

      reducer(undefined, {
        type: `${actionName}_FAILURE`,
        payload: error
      });

      expect(selector(failedState).loading).toEqual(false);
      expect(selector(failedState).value).toEqual([]);
      expect(selector(failedState).error).toEqual(error);
    });
  });
};
