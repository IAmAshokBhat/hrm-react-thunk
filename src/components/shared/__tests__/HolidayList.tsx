// import { screen, waitFor, within } from '@testing-library/react';
import { TAnyObject } from '../../../constants';
import { renderWithProvider } from '../../../testUtils';
import { defaultState } from '../../__tests__/fixtures';
import { HolidayList } from '../HolidayList';

describe('Holiday List component', () => {
  const generateNode = (initialStoreState: TAnyObject = defaultState) =>
    renderWithProvider(<HolidayList />, initialStoreState);

  it('should render and match snapshot', () => {
    const { container } = generateNode();

    expect(container).toMatchSnapshot();
  });
});
