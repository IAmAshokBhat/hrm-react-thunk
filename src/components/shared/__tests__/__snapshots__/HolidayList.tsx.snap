// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Holiday List component should render and match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableCell-root MuiTableCell-sizeMedium css-sow7bl-MuiPaper-root-MuiTableCell-root"
    width="100%"
  >
    <table
      aria-label="holiday table"
      class="MuiTable-root css-1y3zdhj-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
            scope="col"
          >
            Date
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
            scope="col"
          >
            Festival Name
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            01/01/2022
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            New Years day
          </th>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            14/01/2022
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            Sankranthi
          </th>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            26/01/2022
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            Republic Day
          </th>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            17/03/2022
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            Holi
          </th>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            15/04/2022
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            Good Friday
          </th>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            03/05/2022
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            Khutba - E - Ramzan
          </th>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            15/08/2022
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            Independence Day
          </th>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            31/08/2022
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            Ganesha Festival
          </th>
        </tr>
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            05/10/2022
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            scope="row"
          >
            Dasara / Vijayadashmi
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
